kimchi/src/circuits/lookup/lookups.rs

```diff
@@ -204,7 +204,7 @@ impl LookupInfo {
         &self,
         domain: &EvaluationDomains<F>,
         gates: &[CircuitGate<F>],
-    ) -> (LookupSelectors<Evaluations<F>>, Vec<LookupTable<F>>) {
+    ) -> (LookupSelectors<Evaluations<F>>, HashSet<LookupTable<F>>) {
```

Comment:
`HashSet` ordering is non-deterministic (and we probably shouldn't be hashing a whole lookup table). Revert.

Reply:
Hashset order being nondeterministic is true, but the problem is deeper then: we build tables right now on the result of this hashset converted to a vector: we return `gate_tables.iter().convert()`, where `gate_tables` is a hashset. What I did instead is to return it directly without converting to vec, because it might be more convenient to check for duplicates.

The question is: how do you want tables sorted between each other, and is there any difference at all?